src/ctap/crypto_wrapper.rs

```diff
@@ -269,7 +351,14 @@ pub fn decrypt_credential_source(
     if rp_id_hash != &decrypted_id[32..] {
         return Ok(None);
     }
-    let sk_option = PrivateKey::new_ecdsa_from_bytes(&decrypted_id[..32]);
+    let sk_option = match algorithm {
+        ES256_ALGORITHM => PrivateKey::new_ecdsa_from_bytes(&decrypted_id[..32]),
+        #[cfg(feature = "ed25519")]
+        EDDSA_ALGORITHM => PrivateKey::new_ed25519_from_bytes(&decrypted_id[..32]),
+        #[cfg(not(feature = "ed25519"))]
+        EDDSA_ALGORITHM => return Ok(None),
```

Comment:
Why do we need those 2 lines? They're part of the default branch AFAICT.

Reply:
Fixed